src/lwt/win/lwt_unix_impl.ml

```diff
@@ -351,7 +354,10 @@ if !d then Format.eprintf "Start reading@.";
 if !d then Format.eprintf "Reading started@.";
   res
 
-let write ch s pos len =
+let read ch s pos len =
+  _read_string ch (Bytes.to_string s) pos len
```

Comment:
This doesn't make sense to me... I believe `read` should not be touched here, and should directly take a bytes as argument.

Reply:
... see reponse to `write` below. I'll look at it and see if I can refactor it to a form more in line with your suggestion.